### DISCUSSION TITLE
Define max_value for extraAttributes?

### DISCUSSION THREAD

**Discussion by j-greig** [nftchef/art-engine#288] [Help + Q&A] [answered]
Is it possible to set a `max_value` for randomised values created using `extraAttributes`?

As per the metadata standards on OpenSea:  "_Adding an optional max_value sets a ceiling for a numerical trait's possible values. It defaults to the maximum that OpenSea has seen so far on the assets on your contract. If you set a max_value, make sure not to pass in a higher value_."

https://docs.opensea.io/docs/metadata-standards#numeric-traits

In our case we'd set max_value to 100 and be generating a number between 0 and 100.

Thanks!  :)

**Comment by nftchef** [comment, marked as answer]
Yes, the `extraAttribute` config is open to static data or functions.

In this case, `max_value` would be a fixed values. something like 

```
const extraAttributes = () => [
  {
    trait_type: "Aqua Power",
    display_type: "boost_number",
    value: (Math.random() * 100).toFixed(0),
    max_value: 100,
  },
]
```

Will return an attribute like the following:
```
   {
      "trait_type": "Aqua Power",
      "display_type": "boost_number",
      "value": 79,
      "max_value": 100
    }
```

**Reply by j-greig** [reply]
Perfect, thank you for this :)
